test/test_helper.exs

```diff
@@ -137,6 +137,37 @@ defmodule Phoenix.LiveDashboardTest.Endpoint do
   plug Phoenix.LiveDashboardTest.Router
 end
 
+defmodule Phoenix.LiveDashboard.SystemInfoTest.ProcessFilter do
+  @behaviour Phoenix.LiveDashboard.PageFilter
+
+  @impl true
+  def list() do
+    ["All", "Registered", "Phoenix"]
+  end
+
+  @impl true
+  def default_filter() do
+    "Phoenix"
+  end
+
+  def filter("Registered") do
+    Process.registered() |> Enum.map(fn name -> Process.whereis(name) end)
+  end
+
+  @impl true
+  def filter("All") do
+    Process.list()
+  end
+
+  def filter("Phoenix") do
+    Process.registered()
+    |> Enum.flat_map(fn name ->
+      (String.contains?(to_string(name), "Phoenix") &&
+         [Process.whereis(name)]) || []
+    end)
+  end
+end
```

Comment:
Maybe for the test, we don't need a example "so complex", we could return the current pid, or maybe the init process, or a specific application pid. Not important, anyway.

Reply:
I was thinking that it is important to show how to implement the functionality and be able to test it at the same time :-)